### ISSUE TITLE
JIT isn't recognized on iOS <14

### ISSUE THREAD

**Issue by fomalsd** [flyinghead/flycast#538]
Platform / OS / Hardware: `iPadOS 13.4`

Installed via AltStore 1.3.6 + AltServer 1.5b8

Github hash: 20571c5a

Hardware: `iPad Pro 11 2018`

**_Description of the Issue_**
Flycast thinks JIT should be enabled remotely even for older iOS versions that don't need that at all:
`JIT Status: Failed: Error Domain=com.rileytestut.AltServer Code=3 "AltServer could not find this device."`

Other apps that use AltJIT, like [Play!](https://github.com/jpd002/Play-) and [UTM](https://github.com/utmapp/UTM) start up with JIT already enabled.

**_Screenshots_**

![IMG](https://user-images.githubusercontent.com/10403536/155118220-d24dd69f-7e20-4681-9c99-9273ade871ad.jpeg)

**Comment by flyinghead** [comment]
This message only says that it couldn't connect to AltServer. It doesn't mean that JIT isn't enabled (or that JIT is needed on this iOS version). If you enable JIT for Flycast manually through AltServer, this message will still be displayed.

Does Flycast work on this version of iOS?

**Comment by vkedwardli** [comment]
For iOS 13.4, you still need to enable JIT locally thru the psychicpaper exploit, I'm not sure if it is supported in the source code level.

**Comment by fomalsd** [comment]
@flyinghead by default it crashes when trying to run any game (even in Interpreter mode).
If I connect iPad by wire and select "Enable JIT" in AltServer both Dynarec and Interpreter start working, games run fine.
But from what I understand AltKit should handle this automatically, since I don't have to enable JIT manually in other projects that utilize AltKit

**Comment by Spidy123222** [comment]
> For iOS 13.4, you still need to enable JIT locally thru the psychicpaper exploit, I'm not sure if it is supported in the source code level.

The same method JitStreamer and altserver uses still works down to plain iOS 13 and phycicpaper isn't needed.

**Comment by vkedwardli** [comment]
@Spidy123222 yes I'm referring to the fact that apps using the psychicpaper create an illusion that iOS 13 have built-in JIT support. And you are right that using external tools can enable JIT on iOS 13 also.

**Comment by flyinghead** [comment]
@fomalsd You shouldn't have to enable JIT manually. I did a quick test by installing Flycast through the AltStore trusted source and it automatically enables JIT at startup by contacting AltServer.
![image](https://user-images.githubusercontent.com/16261910/169701679-88be284f-3df8-410f-9df5-7d6bfef8a375.png)

I'm not sure it works when sideloading though. It wasn't working in the past but it may have been fixed in recent AltStore versions.
The `AltServer could not find this device.` message usually means that AltStore didn't update the device UDID at Flycast install time.

**Comment by fomalsd** [comment]
@flyinghead just tried the most recent combo of AltServer 1.5 + AltStore 1.5 + Flycast 1.3 3d5fd612 from trusted repo
– it still crashes on game launch on iPadOS 13.4
JIT status states the following:
`Failed: Error Domain=com.rileytestut.AltServer Code=-1 "(null)" UserInfo={NSUnderlyingError=0x280fa72d0 {Error Domain=AltServer.DeveloperDiskError Code=0 "(null)"}}`

![IMG_3428](https://user-images.githubusercontent.com/10403536/170744081-3d38377e-c9ca-4b67-82df-bce99b44fbfd.PNG)

**Comment by flyinghead** [comment]
This looks like an error returned by AltServer. Does manual JIT enabling works wih AltServer?

**Comment by fomalsd** [comment]
No, actually it doesn't. After a while AltServer says 
`The URL to download the Developer disk image could not be determined.` 
for both wired and wireless connections
<img width="372" alt="Scr 2022-05-27 в 21 18 49" src="https://user-images.githubusercontent.com/10403536/170758705-1f1f9f59-8916-4199-aab7-d9e7db416712.png">

But it shouldn't need AltServer for that at all since on iOS 13.4 JIT can be enabled on-device with just the AltStore app using psychicpaper exploit

**Comment by vkedwardli** [comment]
I'm not sure if psychicpaper is still being supported by AltStore / AltKit?

https://github.com/rileytestut/AltKit/issues/7 (oh hi~)
https://github.com/rileytestut/AltStore/issues/613

**Comment by Spidy123222** [comment]
> I'm not sure if psychicpaper is still being supported by AltStore / AltKit?
> 
> 
> 
> https://github.com/rileytestut/AltKit/issues/7 (oh hi~)
> 
> https://github.com/rileytestut/AltStore/issues/613

That is correct and most of the programs that used it before no longer uses it or got broken from updates and was patched after iOS 13.4.1. You will still get a prompt to use it on AltStore app but it will not work on 13.4.1 and below. I actually personally used it until it didn't work.

Altkit isn't phycicpaper, Altkit is just a bunch of tools developers can use but I'm pretty sure your trying to mention altjit function in altkit where it activates a debugger to the app which isn't exclusive to AltStore and never has been. Other programs like JitStreamer,jit workaround and jitterbug can also activate debuggers to use for apps and Xcode has been used before all these.

**Comment by Spidy123222** [comment]
> No, actually it doesn't. After a while AltServer says 
> 
> `The URL to download the Developer disk image could not be determined.` 
> 
> for both wired and wireless connections
> 
> <img width="372" alt="Scr 2022-05-27 в 21 18 49" src="https://user-images.githubusercontent.com/10403536/170758705-1f1f9f59-8916-4199-aab7-d9e7db416712.png">
> 
> 
> 
> But it shouldn't need AltServer for that at all since on iOS 13.4 JIT can be enabled on-device with just the AltStore app using psychicpaper exploit

This issue has been reported that altserver not finding irl for the developerdiskimage. For now I'd suggest using a different program to debug flycast to see if it's just AltStore. I've had quite a bit of unreliability using altjit's automations.

**Comment by fomalsd** [comment]
I have since upgraded to iPadOS 15, but this issue might still be relevant for people on older iOS versions, so leaving it open

**Comment by flyinghead** [comment]
iOS is no longer supported
